dns_test.go

```diff
@@ -168,6 +170,16 @@ func TestNoRdataPack(t *testing.T) {
 }
 
 func TestNoRdataUnpack(t *testing.T) {
+	// TODO(tmthrgd): This test isn't correct and we sometimes don't correctly
+	// pack the messages. It currently succeeds as we explicitly handle certain
+	// broken records in their unpack method, but we shouldn't be. See the TODO
+	// in msg_generate.go. In fact we don't actually produce RRs with an
+	// RDLENGTH of zero, but rather one that just happens to contain mostly
+	// nothing. Further this isn't even how we ourselves construct dynamic
+	// update messages, in update.go we use ANY for when producing an update
+	// message. This correctly produces an record with a type, class, TTL, etc.
```

Comment:
OMFG, yes of course., I can't gather this from your comment (And I'm too lazy to look), but are you saying that `example.org IN A` is made via ANY and fiddled with to make into an A?

if the `type == ` line a few lines below is indeed proof that this was always hacky

Reply:
Yeah basically that. `update.go` uses the `ANY` struct as an explicitly generic RR that has no RDATA. So that already works today, but we also try to handle the zero value of the different RRs, and that doesn't actually work.

Reply:
Ok. I'll take a look separately from this pr to see what can be done here.  It would be nice if the following can be used to make an empty rdata RR:
* NewRR
* &dns.A{}
* and/or setting rdlength to 0

It think it's currently _assumed_ this sorta works today